### ISSUE TITLE
[navicli] remove interactive prompting

### ISSUE THREAD

**Issue by bmr-cymru** [sosreport/sos#405]
```
[plugin:navicli] added copyspec '/etc/Navisphere/agent.config'
[plugin:navicli] added copyspec '/etc/Navisphere/Navimon.cfg'
[plugin:navicli] added copyspec '/etc/Navisphere/Quietmode.cfg'
[plugin:navicli] added copyspec '/etc/Navisphere/messages/[a-z]*'
[plugin:navicli] added copyspec '/etc/Navisphere/log/[a-z]*'
CLARiiON SP IP Address or [Enter] to exit: 
```

Plugins shouldn't be open-coding prompts to the user. There should either be a Plugin API for raising a prompt (possibly combined with the options interface - e.g. an unset option could prompt the user if `prompt=True` is passed) or no interactive prompts after the initial start up.

**Comment by HarryRybacki** [comment]
I believe this can be avoided by using the [``--batch`` flag](https://github.com/sosreport/sos/blob/master/sos/sosreport.py#L1185-L1195).

**Comment by bmr-cymru** [comment]
> I believe this can be avoided by using the --batch flag.

Have you actually tried this? Or even read the code?

Where are you seeing anything testing for the state of `--batch` in this plugin?

(hint: there isn't currently any way for plugins to know this, without digging through SoSOptions directly).

**Comment by bmr-cymru** [comment]
```
[root@localhost ~]# sosreport --batch -e navicli -o navicli

sosreport (version 3.2.0a1)

This command will collect system configuration and diagnostic
information from this Fedora system. An archive containing the collected
information will be generated in /var/tmp/sos.a_v0gjl2.

For more information on the Fedora Project visit:

  https://fedoraproject.org/

The generated archive may contain data considered sensitive and its
content should be reviewed by the originating organization before being
passed to any third party.

No changes will be made to system configuration.


 Setting up archive ...
 Setting up plugins ...
CLARiiON SP IP Address or [Enter] to exit: 
```

See what I mean?

**Comment by HarryRybacki** [comment]
I've tested it locally to collect sosreports [with ansible](https://review.gerrithub.io/#/c/298299/1/tasks/collect.yml). I guess I don't understand exactly what you're looking for -- a *non-interactive* mode or a *silent* mode?

**Comment by bmr-cymru** [comment]
I think you've misunderstood the problem; this issue documents a bug in the `navicli` plugin, in that it attempts to prompt the user for input, regardless of the setting of the `--batch` command line switch.

This is a problem because the definition of `--batch` is:

```
       --batch
              Generate archive without prompting for interactive input.
```

That is the current definition of `--batch` and the behaviour of `navicli` violates this; it has been low priority for us to address in a release since the set of users who have EMC storage and the `NaviCLI` bits installed is relatively small and we've received no complaints from actual end-users in relation to this (the problem was noted during refactoring and this issue filed to keep track of it; really, this should be split into additional issues to cover the API work, and then once that is done and `navicli` uses it, this issue can also be closed).

> I guess I don't understand exactly what you're looking for -- a non-interactive mode or a silent mode?

We have both - interactive prompting and logging output are controlled separately in sos - if you want a silent run, use `--quiet`, to disable interactive prompting, use `--batch`. The issue here is that this plugin ignores the latter switch.

**Comment by bmr-cymru** [comment]
> paste (unformatted) of the ansible task output.

The `navicli` plugin did not execute during this run (most likely you do not have an executable named `navicli` anywhere in path - you can bypass this plugin enabled check by passing `-e navicli` for testing if you wish, as shown a couple of comments back).

**Comment by HarryRybacki** [comment]
Ah, thanks @bmr-cymru, I did miss that you were referring to the plugin causing the breakages.
